prov/sockets/src/sock_progress.c

```diff
@@ -2456,6 +2456,10 @@ static int sock_pe_progress_rx_ep(struct sock_pe *pe, struct sock_ep_attr *ep_at
 	fastlock_acquire(&map->lock);
 	for (i = 0; i < num_fds; i++) {
 		fd = sock_epoll_get_fd_at_index(&map->epoll_set, i);
+#ifdef _WIN32
+		if(fd == -1) /* failed to lookup fd due to connection failures */
+			continue;
+#endif
```

Comment:
@jithinjosepkl - please review this part.

Why is this check only needed on windows?  Can it always just be there?